agent/xds/delta.go

```diff
@@ -679,12 +750,18 @@ func (t *xDSDeltaType) createDeltaResponse(
 	}
 	realUpdates := make(map[string]PendingUpdate)
 	for name, obj := range updates {
-		if obj.Version == "" {
+		if obj.Remove {
+			if obj.Version != "" {
```

Comment:
I can't find a path where these panics could happen. If its really a concern, maybe add a validator fn to PendingUpdate?

Reply:
I had put these panics in to ensure I got my refactor correct. I just didn't circle back to strike them. Doing that now.